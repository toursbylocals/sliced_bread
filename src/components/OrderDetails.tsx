import type { FC } from 'react';

import type { ApiOrderDetails } from '@/api/order/details/[order]';
import Link from 'next/link';

export type OrderDetailsProps = {
  orderDetails: ApiOrderDetails | undefined;
};

export const OrderDetails: FC<OrderDetailsProps> = ({ orderDetails }) => {
  if (!orderDetails) {
    return <>Order not found.</>;
  }

  return (
    <>
      <h2 className="text-2xl font-semibold text-gray-800 mb-4">
        Order Summary
      </h2>
      <div className="space-y-4 mb-8">
        <div className="flex justify-between">
          <span className="font-semibold text-gray-600">Order ID:</span>
          <span className="text-gray-800">{orderDetails.orderId}</span>
        </div>
        <div className="flex justify-between">
          <span className="font-semibold text-gray-600">Customer Name:</span>
          <span className="text-gray-800">{orderDetails.customerName}</span>
        </div>
        <div className="flex justify-between">
          <span className="font-semibold text-gray-600">City:</span>
          <span className="text-gray-800">{orderDetails.city}</span>
        </div>
        <div className="flex justify-between">
          <span className="font-semibold text-gray-600">Province/State:</span>
          <span className="text-gray-800">{orderDetails.provinceState}</span>
        </div>
        <div className="flex justify-between">
          <span className="font-semibold text-gray-600">Country:</span>
          <span className="text-gray-800">{orderDetails.country}</span>
        </div>
        <div className="flex justify-between">
          <span className="font-semibold text-gray-600">Number of Drinks:</span>
          <span className="text-gray-800">{orderDetails.numberOfDrinks}</span>
        </div>
      </div>
      <Link className="text-blue-600" href={'/'}>
        Go back to order form {'>'}
      </Link>
    </>
  );
};
