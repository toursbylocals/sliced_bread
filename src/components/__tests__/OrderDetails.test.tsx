import '@testing-library/jest-dom';

import { render, screen } from '@testing-library/react';

import type { ApiOrderDetails } from '@/api/order/details/[order]';

import { OrderDetails } from '../OrderDetails';

// Mock data for order details
const mockOrderDetails: ApiOrderDetails = {
  orderId: '12345',
  customerName: 'John Doe',
  city: 'New York',
  provinceState: 'NY',
  country: 'USA',
  numberOfDrinks: 3,
};

describe('OrderDetails Component', () => {
  it('renders order details correctly when orderDetails is provided', () => {
    render(<OrderDetails orderDetails={mockOrderDetails} />);

    // Check that the title "Order Summary" is rendered
    expect(screen.getByText('Order Summary')).toBeInTheDocument();

    // Check that each field in the orderDetails is rendered correctly
    expect(screen.getByText('Order ID:')).toBeInTheDocument();
    expect(screen.getByText(mockOrderDetails.orderId)).toBeInTheDocument();

    expect(screen.getByText('Customer Name:')).toBeInTheDocument();
    expect(screen.getByText(mockOrderDetails.customerName)).toBeInTheDocument();

    expect(screen.getByText('City:')).toBeInTheDocument();
    expect(screen.getByText(mockOrderDetails.city)).toBeInTheDocument();

    expect(screen.getByText('Province/State:')).toBeInTheDocument();
    expect(
      screen.getByText(mockOrderDetails.provinceState),
    ).toBeInTheDocument();

    expect(screen.getByText('Country:')).toBeInTheDocument();
    expect(screen.getByText(mockOrderDetails.country)).toBeInTheDocument();

    expect(screen.getByText('Number of Drinks:')).toBeInTheDocument();
    expect(
      screen.getByText(mockOrderDetails.numberOfDrinks.toString()),
    ).toBeInTheDocument();
  });

  it('displays "Order not found." when orderDetails is not provided', () => {
    render(<OrderDetails orderDetails={undefined} />);

    // Check that the message "Order not found." is displayed
    expect(screen.getByText('Order not found.')).toBeInTheDocument();
  });

  it('contains a link that navigates to the home page', () => {
    render(<OrderDetails orderDetails={mockOrderDetails} />);

    // Check that the link to go back to the order form is present
    const linkElement = screen.getByRole('link', {
      name: /Go back to order form/i,
    });

    // Ensure the link has the correct href
    expect(linkElement).toHaveAttribute('href', '/');
  });

  it('matches the snapshot when orderDetails are provided', () => {
    const { asFragment } = render(
      <OrderDetails orderDetails={mockOrderDetails} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches the snapshot when no orderDetails are provided', () => {
    const { asFragment } = render(<OrderDetails orderDetails={undefined} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
