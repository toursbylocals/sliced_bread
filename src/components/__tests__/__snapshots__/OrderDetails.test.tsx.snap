// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderDetails Component matches the snapshot when no orderDetails are provided 1`] = `
<DocumentFragment>
  Order not found.
</DocumentFragment>
`;

exports[`OrderDetails Component matches the snapshot when orderDetails are provided 1`] = `
<DocumentFragment>
  <h2
    class="text-2xl font-semibold text-gray-800 mb-4"
  >
    Order Summary
  </h2>
  <div
    class="space-y-4 mb-8"
  >
    <div
      class="flex justify-between"
    >
      <span
        class="font-semibold text-gray-600"
      >
        Order ID:
      </span>
      <span
        class="text-gray-800"
      >
        12345
      </span>
    </div>
    <div
      class="flex justify-between"
    >
      <span
        class="font-semibold text-gray-600"
      >
        Customer Name:
      </span>
      <span
        class="text-gray-800"
      >
        John Doe
      </span>
    </div>
    <div
      class="flex justify-between"
    >
      <span
        class="font-semibold text-gray-600"
      >
        City:
      </span>
      <span
        class="text-gray-800"
      >
        New York
      </span>
    </div>
    <div
      class="flex justify-between"
    >
      <span
        class="font-semibold text-gray-600"
      >
        Province/State:
      </span>
      <span
        class="text-gray-800"
      >
        NY
      </span>
    </div>
    <div
      class="flex justify-between"
    >
      <span
        class="font-semibold text-gray-600"
      >
        Country:
      </span>
      <span
        class="text-gray-800"
      >
        USA
      </span>
    </div>
    <div
      class="flex justify-between"
    >
      <span
        class="font-semibold text-gray-600"
      >
        Number of Drinks:
      </span>
      <span
        class="text-gray-800"
      >
        3
      </span>
    </div>
  </div>
  <a
    class="text-blue-600"
    href="/"
  >
    Go back to order form &gt;
  </a>
</DocumentFragment>
`;
